File: read_pdf.js
const fs = require('fs');
const pdf = require('pdf-parse');

const dataBuffer = fs.readFileSync('the branding method (carolina kairos) (Z-Library).pdf');
const parseFn = typeof pdf === 'function' ? pdf : pdf.default || pdf;

parseFn(dataBuffer).then(function(data) {
    fs.writeFileSync('book_text.txt', data.text);
    console.log('PDF text extracted to book_text.txt (Length: ' + data.text.length + ' chars)');
}).catch(function(error){
    console.error('Error parsing PDF:', error);
});

File: read_pdf.py
import sys
try:
    from pypdf import PdfReader
    reader = PdfReader("the branding method (carolina kairos) (Z-Library).pdf")
    text = ""
    for page in reader.pages:
        extracted = page.extract_text()
        if extracted:
            text += extracted + "\n"
    with open("book_text.txt", "w", encoding="utf-8") as f:
        f.write(text)
    print("PDF successfully parsed, length: " + str(len(text)))
except Exception as e:
    print("Error:", e)
